src/Application/ConfigAuthorizer.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Application;
+
+use MediaWiki\Page\ProperPageIdentity;
+use MediaWiki\User\User;
+
+class ConfigAuthorizer {
+
+	public function __construct(
+		private readonly bool $enableWikiConfig,
```

Comment:
Minor nitpick: `$enableWikiConfig` sounds like the name of a method/action. `$wikiConfigIsEnabled` sounds more like a boolean.